spec/features/shared_examples/cherrypicking.rb

```diff
@@ -237,14 +237,27 @@
 
             click_link("Download #{robot.name} File")
 
+            current_destination_plate = Plate.find_by_barcode(destination_barcode)
+
             # robot file generation differs by generator
             case robot.generation_behaviour_property.value
             when 'Hamilton'
               # for Robot::Generator::Hamilton
               generated_file = DownloadHelpers.downloaded_file("#{batch_id}_batch_#{destination_barcode}_#{pick_number_index}.csv")
               generated_lines = generated_file.lines
 
-              # optionally if an expected file was supplied. compare it to the result
+              # check generated lines match expected by calculation - headers row + one row per sample/control transfer
+              expected_num_lines = current_destination_plate.wells.map(&:transfer_requests_as_target).count + 1
```

Comment:
Is that +1 the header?

Reply:
Yes a single header row.
Tecan files have additional info at the top of the file e.g. user/date.
Hamilton files are just csv's with a column headers row at the top and one row per transfer..